public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -245,14 +249,7 @@ export class CloudWatchDatasource
     );
   };
 
-  filterQuery(query: CloudWatchQuery): boolean {
-    if (isCloudWatchLogsQuery(query)) {
-      return !!query.logGroupNames?.length;
-    } else if (isCloudWatchAnnotationQuery(query)) {
```

Comment:
I think you have removed the condition here to filter a query if it's a logs query and it doesn't have group names.

Reply:
Yeah I previously thought that the `filterQuery` method was executed for all queries. Turns out it's only for metric queries, so that's why I renamed the method to `filterMetricsQuery`. 

Logs queries are filtered in the [`handleLogQueries` method](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/datasource.ts#L175), so should be fine. Sorry should have provided this context in PR description. 

It's a bit confusing that this data source file handles so many query types. I'm not a huge fan of the "pseudo data source" concept we have in AM, but maybe it's about time to introduce it also for CW so that it easier to overview the execution flow (which filtering is a part of).

Reply:
got it, yeah, seems a bit difficult to follow